atom/browser/web_contents_preferences.cc

```diff
@@ -155,8 +156,19 @@ void WebContentsPreferences::AppendExtraCommandLineSwitches(
     command_line->AppendSwitchASCII(::switches::kEnableBlinkFeatures,
                                     blink_features);
 
-  // The initial visibility state.
   NativeWindow* window = NativeWindow::FromWebContents(web_contents);
+
+  // Inherit initial visibilty state from parent window in webviews
+  if (guest_instance_id && !window) {
+    auto manager = WebViewManager::GetWebViewManager(web_contents);
+    if (manager) {
+      content::WebContents* embedder = manager->GetEmbedder(guest_instance_id);
```

Comment:
@zcbenz Thanks for pointing out `web_view_manager`. I added a new API to get the embedder web contents for a guest instance id which allows window lookup.

Is this in line with what you were suggesting?

I've removed the passing of `hiddenPage` in the params via this approach, thanks for mentioning that, I think this approach is cleaner.